File: LEETCODE/ARRAY/167. Two Sum II - Input Array Is Sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        # #T.C-> O(N)
        hashmap={}
        for i in range(len(numbers)):
            val=target-numbers[i]
            if val in hashmap:
                return [hashmap[val],i+1]
            else:
                hashmap[numbers[i]]=i+1
        
        
        #another solution
        # a+b=k
        # b=k-a
        #bruteforce
        # for i in range(len(numbers)-1):
        #     val=target-numbers[i]
        #     for j in range(i+1,len(numbers)):
        #         if numbers[j]==val:
        #             return [i+1,j+1]
        
        
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        

File: LEETCODE/ARRAY/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    // using the concept of Binary Search
    
    public int[] twoSum(int[] numbers, int target) {
        int low=0;
        int high=numbers.length-1;
        int[] arr=new int[2];
        while(low<high){
            int sum=numbers[low]+numbers[high];
            if(sum==target){
                arr[0]=low+1;
                arr[1]=high+1;
                break;
            }
            else if(sum<target){
                low++;
            }
            else{
                high--;
            }
        }
        return arr;
    }
}
